src/app/modules/admin-feedback/__tests__/admin_feedback.service.spec.ts

```diff
@@ -0,0 +1,246 @@
+
+const AdminFeedbackModel = getAdminFeedbackModel(mongoose)
+
+describe('feedback.service', () => {
+  beforeAll(() => dbHandler.connect())
+  afterAll(() => dbHandler.closeDatabase())
+
+  describe('insertAdminFeedback', () => {
+    beforeEach(async () => {
+      await dbHandler.clearCollection(AdminFeedbackModel.collection.name)
+    })
+    afterEach(() => jest.clearAllMocks())
+
+    const MOCK_USER_ID = new ObjectId().toHexString()
+    const MOCK_RATING = 1
+    const MOCK_COMMENT = 'mock comment'
+    const MOCK_ADMIN_FEEDBACK = new AdminFeedbackModel({
+      userId: MOCK_USER_ID,
+      rating: MOCK_RATING,
+      comment: MOCK_COMMENT,
+    })
+
+    it('should return Admin Feedback document on successful insertion with comment', async () => {
+      // Arrange
+      // Mock success.
+      const insertSpy = jest
+        .spyOn(AdminFeedbackModel, 'create')
+        // @ts-ignore
+        .mockResolvedValueOnce(MOCK_ADMIN_FEEDBACK)
+
+      // Act
+      const actualResult = await AdminFeedbackService.insertAdminFeedback({
+        userId: MOCK_USER_ID,
+        rating: MOCK_RATING,
+        comment: MOCK_COMMENT,
+      })
+
+      // Assert
+      expect(insertSpy).toHaveBeenCalledTimes(1)
+      expect(actualResult.isOk()).toEqual(true)
+      expect(actualResult._unsafeUnwrap()).toEqual(MOCK_ADMIN_FEEDBACK)
+    })
+
+    it('should return Admin Feedback document on successful insertion without comment', async () => {
+      const mock_feedback_no_comment = new AdminFeedbackModel({
+        userId: MOCK_USER_ID,
+        rating: MOCK_RATING,
+      })
+      // Arrange
+      // Mock success.
+      const insertSpy = jest
+        .spyOn(AdminFeedbackModel, 'create')
+        // @ts-ignore
+        .mockResolvedValueOnce(mock_feedback_no_comment)
+
+      // Act
+      const actualResult = await AdminFeedbackService.insertAdminFeedback({
+        userId: MOCK_USER_ID,
+        rating: MOCK_RATING,
+      })
+
+      // Assert
+      expect(insertSpy).toHaveBeenCalledTimes(1)
+      expect(actualResult.isOk()).toEqual(true)
+      expect(actualResult._unsafeUnwrap()).toEqual(mock_feedback_no_comment)
+    })
+
+    it('should return DatabaseError when error occurs whilst inserting feedback', async () => {
+      // Arrange
+      // Mock failure
+      const insertSpy = jest
+        .spyOn(AdminFeedbackModel, 'create')
+        // @ts-ignore
+        .mockRejectedValueOnce(new Error('some error'))
+
+      // Act
+      const actualResult = await AdminFeedbackService.insertAdminFeedback({
+        userId: MOCK_USER_ID,
+        rating: MOCK_RATING,
+      })
+
+      // Assert
+      expect(insertSpy).toHaveBeenCalledTimes(1)
+      expect(actualResult.isErr()).toEqual(true)
+      expect(actualResult._unsafeUnwrapErr()).toBeInstanceOf(DatabaseError)
+    })
+  })
+
+  describe('updateAdminFeedback', () => {
+    const MOCK_FEEDBACK_ID = new ObjectId().toHexString()
```

Comment:
should we add spy to adminFeedbackModel?

Reply:
I think there isn't a need? As we are able to verify the interaction with the model through the results. And for most of the update calls, I will subsequently make another call to retrieve the updated results, so if the updated result has the intended behaviour it should be robust enough

The spy doesn't really do much and just check for whether it is called 😅 and I don't want to mock it so as to test mongoose's static behaviour too

Reply:
what bout the case where no comment and rating, dont we need to verify that adminFeedbackModel is not called?

Reply:
Oh yes thats big brain thanks @mayying [6472c7d](https://github.com/opengovsg/FormSG/pull/6454/commits/6472c7d9b9d8f4290a0a0163e0e4fe7b09b78666)